src/Bicep.Core/Analyzers/Linter/Rules/UseResourceIdFunctionsRule.cs

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Bicep.Core.Analyzers.Linter.Common;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Navigation;
+using Bicep.Core.Semantics;
+using Bicep.Core.Syntax;
+using Microsoft.WindowsAzure.ResourceStack.Common.Extensions;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class UseResourceIdFunctionsRule : LinterRuleBase
+    {
+        public new const string Code = "use-resource-id-functions";
+
+        public UseResourceIdFunctionsRule() : base(
```

Comment:
I am not seeing warning for below case. Is it expected?
![image](https://user-images.githubusercontent.com/30270536/185661794-7d357ca9-815f-4f34-bfb8-bcd3fad14760.png)

Sample bicep file contents:
```
resource applicationGateway 'Microsoft.Network/applicationGateways@2020-11-01' = {
  name: 'name'
#disable-next-line no-loc-expr-outside-params
  location: resourceGroup().location
  properties: {
    sku: {
      name: 'Standard_Small'
      tier: 'Standard'
      capacity: 'capacity'
    }
    gatewayIPConfigurations: [
      {
        name: 'name'
        properties: {
          subnet: {
            id: '/subscriptions/${subscription().subscriptionId}/resourceGroups/${resourceGroup().name}/providers/Microsoft.Network/networkSecurityGroups/${NSGName}'
          }
        }
      }
    ]
  }
}
```

Reply:
You have an undefined variable.  Fix that it the linter error shows up.  Added a testcase anyway, good scenario.